cfme/physical/physical_chassis.py

```diff
@@ -0,0 +1,294 @@
+# -*- coding: utf-8 -*-
+"""A model of an Infrastructure PhysicalChassis in CFME."""
+import attr
+
+from lxml.html import document_fromstring
+from navmazing import NavigateToAttribute
+from widgetastic_patternfly import Dropdown, Accordion
+from widgetastic.widget import Text, View
+from wrapanapi.entities import PhysicalContainer
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.common import PolicyProfileAssignable, Taggable
+from cfme.exceptions import (
+    ItemNotFound,
+    StatsDoNotMatch,
+    HostStatsNotContains,
+    ProviderHasNoProperty
+)
+from cfme.modeling.base import BaseEntity, BaseCollection
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigate_to, navigator
+from cfme.utils.log import logger
+from cfme.utils.pretty import Pretty
+from cfme.utils.providers import get_crud_by_name
+from cfme.utils.update import Updateable
+from cfme.utils.varmeth import variable
+from widgetastic_manageiq import (
+    BaseEntitiesView,
+    JSBaseEntity,
+    BreadCrumb,
+    ItemsToolBarViewSelector,
+    SummaryTable,
+    ManageIQTree
+)
+
+
+@attr.s
+class PhysicalChassis(BaseEntity, Updateable, Pretty, PolicyProfileAssignable, Taggable):
+    """Model of an Physical Chassis in cfme.
+
+    Args:
+        name: Name of the physical chassis.
+        custom_ident: The custom identifiter.
+
+    """
+    pretty_attrs = ['name', 'custom_ident']
+
+    name = attr.ib()
+    ems_ref = attr.ib(default=None)
+    provider = attr.ib(default=None)
+    custom_ident = attr.ib(default=None)
+    db_id = None
+    mgmt_class = PhysicalContainer
+
+    INVENTORY_TO_MATCH = []
+    STATS_TO_MATCH = []
+
+    def load_details(self, refresh=False):
+        """To be compatible with the Taggable and PolicyProfileAssignable mixins.
+
+        Args:
+            refresh (bool): Whether to perform the page refresh, defaults to False
+        """
+        view = navigate_to(self, "Details")
+        if refresh:
+            view.browser.refresh()
+            view.flush_widget_cache()
+
+    def _execute_button(self, button, option, handle_alert=False):
+        view = navigate_to(self, "Details")
+        view.toolbar.custom_button(button).item_select(option, handle_alert=handle_alert)
+        return view
+
+    @property
+    def exists(self):
+        """Checks if the physical_chassis exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        try:
+            navigate_to(self, 'Details')
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    @variable(alias='ui')
+    def chassis_name(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Chassis name")
```

Comment:
Please see my other comment on this topic, but I think approach with multiple entities is deprechated. Try go with:

```suggestion
        return view.entities.summary("Properties").get_text_of("Chassis name")
```

Reply:
👍